core/state/commonstate/deprecated_state_test.go

```diff
@@ -0,0 +1,27 @@
+package commonstate
+
+import (
+	"testing"
+
+	"github.com/NethermindEth/juno/core"
+	"github.com/NethermindEth/juno/db/memory"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestDeprecatedStateAdapter(t *testing.T) {
+	testDB := memory.New()
+	txn := testDB.NewIndexedBatch()
+
+	state := core.NewState(txn)
+	deprecatedStateAdapter := NewDeprecatedStateAdapter(state)
+	assert.NotNil(t, deprecatedStateAdapter)
+}
+
+func TestDeprecatedStateReaderAdapter(t *testing.T) {
+	testDB := memory.New()
+	txn := testDB.NewIndexedBatch()
+
+	state := core.NewState(txn)
+	deprecatedStateAdapter := NewDeprecatedStateReaderAdapter(state)
+	assert.NotNil(t, deprecatedStateAdapter)
+}
```

Comment:
Another reason why I advocate removing the adapter is because of missing tests. We have several groups of functions sharing the same function signature. For example, `ContractClassHash`, `ContractNonce` and `ContractStorage` all share the same signature `func(addr *felt.Felt) (felt.Felt, error)`. The same thing applies to `...At` and `...Trie` variants. Without tests, there is no way we can guarantee that the adapter's `ContractClassHash` calls underlying `ContractClassHash` instead of `ContractNonce`.